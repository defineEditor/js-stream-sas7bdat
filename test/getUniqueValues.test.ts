import DatasetSas7BDat from '../src/index';
import path from 'path';

test('Get unique values of variables', async () => {
    const filePath = path.join(__dirname, '/data/sample.sas7bdat');
    const data = new DatasetSas7BDat(filePath);
    const values = await data.getUniqueValues({columns: ['NAME', 'AGE', 'WEIGHT'], limit: 20});
    expect(values).toMatchSnapshot();
});

test('Get unique values of variables and do not sort the result', async () => {
    const filePath = path.join(__dirname, '/data/sample.sas7bdat');
    const data = new DatasetSas7BDat(filePath);
    const values = await data.getUniqueValues({columns: ['NAME', 'HEIGHT', 'WEIGHT'], limit: 20, bufferLength: 10, sort: false});
    expect(values).toMatchSnapshot();
});

test('Get unique values with counts', async () => {
    const filePath = path.join(__dirname, '/data/sample.sas7bdat');
    const data = new DatasetSas7BDat(filePath);
    const values = await data.getUniqueValues({columns: ['NAME', 'AGE', 'WEIGHT'],
        limit: 10, addCount: true, sort: false});
    expect(values).toMatchSnapshot();
});