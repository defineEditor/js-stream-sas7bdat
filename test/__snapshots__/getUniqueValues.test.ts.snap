// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get unique values of variables 1`] = `
{
  "Age": {
    "counts": {},
    "values": [
      11,
      12,
      13,
      14,
      15,
      16,
    ],
  },
  "Name": {
    "counts": {},
    "values": [
      "Alfred",
      "Alice",
      "Barbara",
      "Carol",
      "Henry",
      "James",
      "Jane",
      "Janet",
      "Jeffrey",
      "John",
      "Joyce",
      "Judy",
      "Louise",
      "Mary",
      "Philip",
      "Robert",
      "Ronald",
      "Thomas",
      "William",
    ],
  },
  "Weight": {
    "counts": {},
    "values": [
      102.5,
      112,
      112.5,
      128,
      133,
      150,
      50.5,
      77,
      83,
      84,
      84.5,
      85,
      90,
      98,
      99.5,
    ],
  },
}
`;

exports[`Get unique values of variables and do not sort the result 1`] = `
{
  "Height": {
    "counts": {},
    "values": [
      69,
      56.5,
      65.3,
      62.8,
      63.5,
      57.3,
      59.8,
      62.5,
      59,
      51.3,
      64.3,
      56.3,
      66.5,
      72,
      64.8,
      67,
      57.5,
    ],
  },
  "Name": {
    "counts": {},
    "values": [
      "Alfred",
      "Alice",
      "Barbara",
      "Carol",
      "Henry",
      "James",
      "Jane",
      "Janet",
      "Jeffrey",
      "John",
      "Joyce",
      "Judy",
      "Louise",
      "Mary",
      "Philip",
      "Robert",
      "Ronald",
      "Thomas",
      "William",
    ],
  },
  "Weight": {
    "counts": {},
    "values": [
      112.5,
      84,
      98,
      102.5,
      83,
      84.5,
      99.5,
      50.5,
      90,
      77,
      112,
      150,
      128,
      133,
      85,
    ],
  },
}
`;

exports[`Get unique values with counts 1`] = `
{
  "Age": {
    "counts": {
      "11": 2,
      "12": 5,
      "13": 3,
      "14": 4,
      "15": 4,
      "16": 1,
    },
    "values": [
      14,
      13,
      12,
      15,
      11,
      16,
    ],
  },
  "Name": {
    "counts": {
      "Alfred": 1,
      "Alice": 1,
      "Barbara": 1,
      "Carol": 1,
      "Henry": 1,
      "James": 1,
      "Jane": 1,
      "Janet": 1,
      "Jeffrey": 1,
      "John": 1,
      "Joyce": 1,
      "Judy": 1,
      "Louise": 1,
      "Mary": 1,
      "Philip": 1,
      "Robert": 1,
      "Ronald": 1,
      "Thomas": 1,
      "William": 1,
    },
    "values": [
      "Alfred",
      "Alice",
      "Barbara",
      "Carol",
      "Henry",
      "James",
      "Jane",
      "Janet",
      "Jeffrey",
      "John",
    ],
  },
  "Weight": {
    "counts": {
      "102.5": 2,
      "112": 2,
      "112.5": 2,
      "128": 1,
      "133": 1,
      "150": 1,
      "50.5": 1,
      "77": 1,
      "83": 1,
      "84": 2,
      "84.5": 1,
      "85": 1,
      "90": 1,
      "98": 1,
      "99.5": 1,
    },
    "values": [
      112.5,
      84,
      98,
      102.5,
      83,
      84.5,
      99.5,
      50.5,
      90,
      77,
    ],
  },
}
`;
